### PR TITLE
[demo] Add C++ metrics

### LINKED ISSUE #2403: Demo Documentation - Update Currency Service Metrics
### Demo Documentation Update
Whenever https://github.com/open-telemetry/opentelemetry-demo/pull/759 gets merged, the following pages need to be updated:

- [x] https://opentelemetry.io/docs/demo/metric-features/
- [x] https://opentelemetry.io/docs/demo/services/currency/#metrics

### PR DESCRIPTION
**Pull request by julianocosta89**
Fix #2403

Added documentation about C++ metrics, which was introduced to the demo in this PR: https://github.com/open-telemetry/opentelemetry-demo/pull/759

### CODE REVIEW

File: content/en/docs/demo/services/currency.md [hunk 1]
```diff
@@ -137,7 +137,70 @@ creating new spans.
...
+  auto p = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
+  p->AddMetricReader(std::move(reader));
+  metrics_api::Provider::SetMeterProvider(provider);
+}
+```
+
+### Starting IntCounter
+
+A global `currency_counter` variable is created at `main` calling the function
+`initIntCounter` defined at `meter_common.h`.
```
Comment (chalin) [inline content/en/docs/demo/services/currency.md:172]:
```suggestion
`initIntCounter()` defined in `meter_common.h`.
```

File: content/en/docs/demo/services/currency.md [hunk 2]
```diff
@@ -137,7 +137,70 @@ creating new spans.
 
 ## Metrics
 
-TBD
+### Initializing Metrics
+
+The OpenTelemetry `MeterProvider` is initialized from `main` using the `initMeter` function
```
Comment (chalin) [inline content/en/docs/demo/services/currency.md:142]:
```suggestion
The OpenTelemetry `MeterProvider` is initialized from `main()` using the `initMeter()` function
```

File: content/en/docs/demo/services/currency.md [hunk 3]
```diff
@@ -137,7 +137,70 @@ creating new spans.
...
+  auto provider = std::shared_ptr<metrics_api::MeterProvider>(new metric_sdk::MeterProvider());
+  auto p = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
+  p->AddMetricReader(std::move(reader));
+  metrics_api::Provider::SetMeterProvider(provider);
+}
+```
+
+### Starting IntCounter
+
+A global `currency_counter` variable is created at `main` calling the function
```
Comment (chalin) [inline content/en/docs/demo/services/currency.md:171]:
```suggestion
A global `currency_counter` variable is created at `main()` calling the function
```

File: content/en/docs/demo/services/currency.md [hunk 4]
```diff
@@ -137,7 +137,70 @@ creating new spans.
...
+  auto provider = metrics_api::Provider::GetMeterProvider();
+  nostd::shared_ptr<metrics_api::Meter> meter = provider->GetMeter(name, version);
+  auto int_counter = meter->CreateUInt64Counter(counter_name);
+  return int_counter;
+}
+```
+
+### Counting currency conversion requests
+
+The method `CurrencyCounter` is implemented as follows.
```
Comment (chalin) [inline content/en/docs/demo/services/currency.md:187]:
```suggestion
The method `CurrencyCounter()` is implemented as follows:
```

File: content/en/docs/demo/services/currency.md [hunk 5]
```diff
@@ -137,7 +137,70 @@ creating new spans.
...
+```cpp
+void CurrencyCounter(const std::string& currency_code)
+{
+    std::map<std::string, std::string> labels = { {"currency_code", currency_code} };
+    auto labelkv = common::KeyValueIterableView<decltype(labels)>{ labels };
+    currency_counter->Add(1, labelkv);
+}
+```
+
+Every time the function `Convert` is called, the currency code
```
Comment (chalin) [inline content/en/docs/demo/services/currency.md:198]:
```suggestion
Every time the function `Convert()` is called, the currency code
```

**Review by chalin — COMMENTED** [review]
See inline comments, thanks.

**Review by fatsheep9146 — APPROVED** [review]

**Review by svrnm — APPROVED** [review]

**Pull request merged**